### ISSUE TITLE
Security enhancement when following a "redirect"

### ISSUE THREAD

**Issue by mvsantos** [scrapy/scrapy#457]
I believe this is not a bug, but could fit in as a security enhancement.

**TL;DR** When following HTTP redirects, scrapy should only follow http/https requests. Quite possibly similar to (CVE-2009-0037) http://curl.haxx.se/docs/adv_20090303.html

Scenario:
a. A spider that simply stores raw or minimally sanitised data.
b. By default scrapy follows HTTP redirects.
c. The values of both DOWNLOAD_HANDLERS and DOWNLOAD_HANDLERS_BASE are set to their default.

``` python
DOWNLOAD_HANDLERS = {}
DOWNLOAD_HANDLERS_BASE = {
    'file': 'scrapy.core.downloader.handlers.file.FileDownloadHandler',
    'http': 'scrapy.core.downloader.handlers.http.HTTPDownloadHandler',
    'https': 'scrapy.core.downloader.handlers.http.HTTPDownloadHandler',
    's3': 'scrapy.core.downloader.handlers.s3.S3DownloadHandler',
    'ftp': 'scrapy.core.downloader.handlers.ftp.FTPDownloadHandler',
}
```

And let's say a malicious client might issue a redirect such as

```
# ==== .htaccess ====
RewriteEngine On
Redirect 301 /trap.html  file:///etc/passwd
# ==== ENDof .htaccess ====
```

And so, my spider will be "leaking" my own _"/etc/passwd"_ file because Scrapy followed the 301 redirect from _"http:// ... /trap.html"_ to _"file:///etc/passwd"_.

Shouldn't scrapy enforce some sort of protocol restriction so that it only follows the http/https protocols when following redirects?
